elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/engine/StitchList.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.engine;
+
+import com.yahoo.elide.core.EntityDictionary;
+
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Getter;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * An auxiliary class for {@link AbstractEntityHydrator} and is responsible for setting relationship values of an entity
+ * instance.
+ * <p>
+ * {@link StitchList} should not be subclassed.
+ * <p>
+ * Concurrency note: {@link #stitch()} operation generally do not block, so may overlap with update operations (
+ * {@link #todo(Object, String, Object)} and {@link #populateLookup(Class, Map)}). {@link #stitch()} reflects the
+ * results of the most recently completed update operations holding upon their onset. (More formally, an update
+ * operation bears a happens-before relation with a {@link #stitch()} operation.)
+ */
+public final class StitchList {
+
+    /**
+     * A representation of an TODO item in a {@link StitchList}.
+     */
+    @Data
+    public static class Todo {
+        private final Object entityInstance;
+        private final String relationshipName;
+        private final Object foreignKey;
+    }
+
+    /**
+     * Maps an relationship entity class to a map of object ID to object instance.
+     * <p>
+     * For example, [Country.class: [340: Country(id:340), 100: Country(id:100)]]
+     */
+    @Getter(AccessLevel.PRIVATE)
+    private final Map<Class<?>, Map<Object, Object>> objectLookups;
+
+    /**
+     * List of relationships to hydrate
+     */
+    @Getter(AccessLevel.PRIVATE)
+    private final List<Todo> todoList;
+
+    @Getter(AccessLevel.PRIVATE)
+    private final EntityDictionary entityDictionary;
+
+    /**
+     * Constructor.
+     *
+     * @param entityDictionary  An object that sets entity instance values and provides entity metadata info
+     */
+    public StitchList(EntityDictionary entityDictionary) {
+        this.objectLookups = new ConcurrentHashMap<>();
```

Comment:
These classes will always be accessed from a single thread.  I don't think we need the overhead of a concurrent hash map.